### PR TITLE
Display diagnostic in Java file after loading of MP Java project

### LINKED ISSUE #301: Java source code not validated upon start
I noticed that Java source files are not validated when the file is opened.
Seems to be initialization related as if works correctly if you close and reopen the file

### PR DESCRIPTION
**Pull request by angelozerr**
Display diagnostic in Java file after loading of MP Java project

Fixes #301

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by jeffmaury** [comment]
Fixes #301

**Comment by angelozerr** [comment]
As @rgrunber noticed, there is the same problem with inlay hint for properties file. This PR should fix too the problem.

**Comment by angelozerr** [comment]
I would like to investigate if I could write easily some tests with thoses fixes.

File: microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaTextDocuments.java [hunk 1]
```diff
@@ -118,27 +118,66 @@ public void setProjectURI(String projectURI) {
 
 		/**
 		 * Execute the given code only if the Java file belongs to a MicroProfile
-		 * project.
+		 * without waiting for the load of project information. project.
```
Comment (datho7561) [inline microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaTextDocuments.java:121]: Remove " project." from the comment

File: microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaTextDocuments.java [hunk 2]
```diff
@@ -118,27 +118,66 @@ public void setProjectURI(String projectURI) {
...
 		 * Execute the given code only if the Java file belongs to a MicroProfile
-		 * project.
+		 * without waiting for the load of project information. project.
 		 *
 		 * @param <T>          the type to return.
 		 * @param code         the code to execute.
 		 * @param defaultValue the default value to return if the Java file doesn't
 		 *                     belong to a MicroProfile project.
 		 * @return the given code only if the Java file belongs to a MicroProfile
+		 *         project without waiting for the load of project information. project.
```
Comment (datho7561) [inline microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaTextDocuments.java:128]: Remove " project." from the comment

**Review by datho7561 — CHANGES_REQUESTED** [review]
This change is working well. Thanks, Angelo! I noticed a small mistake in some of the comments

File: microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaTextDocuments.java (hunk 2 above)
Comment (angelozerr) [reply microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaTextDocuments.java:128]: fixed

File: microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaTextDocuments.java (hunk 1 above)
Comment (angelozerr) [reply microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaTextDocuments.java]: fixed

**Comment by angelozerr** [comment]
Ok now we have tests with diagnostics and inlay hint when project is loading.

**Review by datho7561 — APPROVED** [review]
Looks good to me. Thanks, Angelo!

**Pull request merged**